File: solutions/javascript/pig-latin/1/pig-latin.js

export const translate = (input) => {
  return input.split(" ").map(word => xlateWord(word)).join(' ');
};


const reconstruct = (word, match) => (
  word.substring(0 + match[1].length) + match[1] + "ay"
)

const xlateWord = (input) => {
  const word = input.toLowerCase();

  const vow = /^([aeiou]|xr|yt)+/;
  const cons = /^([^aeiouy]+)/;
  const consQU = /^([^aeiouy]*qu)/;
  const wordY = /^y/;
  let match = null;

  if ( word.match(vow) ) return word + 'ay';
  if ( match = word.match(consQU) ) return reconstruct(word, match);
  if ( match = word.match(cons) ) return reconstruct(word, match);
  if ( match = word.match(wordY) ) return word.substring(1) + 'yay';

  return '';
}

File: solutions/python/pig-latin/1/pig_latin.py
def translate(text):
    words = text.split()
    pig_latin = []
    for word in words:
        if word[0:2] == 'yt' or word[0:2] == 'xr':
            pig_latin.append(word + 'ay')
        elif word[0:3] == 'sch' or word[0:3] == 'thr' or word[0:3] == 'squ':
            pig_latin.append(word[3:] + word[0:3] + 'ay')
        elif word[0:2] == 'ch' or word[0:2] == 'qu' or word[0:2] == 'th' or word[0:2] == 'rh':
            pig_latin.append(word[2:] + word[0:2] + 'ay')
        elif word[0] in 'aeiou':
            pig_latin.append(word + 'ay')
        else:
            pig_latin.append(word[1:] + word[0] + 'ay')
    return ' '.join(pig_latin)
